scripts/gen_mmu.py

```diff
@@ -0,0 +1,425 @@
+import os
+import sys
+import struct
+import parser
+from collections import namedtuple
+import ctypes
+import argparse
+
+############# global variables
+pd_complete = ''
+inputfile = ''
+outputfile = ''
+list_of_pde = {}
+num_of_regions = 0
+read_buff=''
+
+struct_mmu_regions_tuple = {"start_addr","size","permissions"}
+mmu_region_details = namedtuple("mmu_region_details", "start_addr size pde_index \
+                                pte_valid_addr_start \
+                                pte_valid_addr_end \
+                                permissions")
+
+page_tables_list = []
+PD_start_addr = 0
+validation_issue_memory_overlap = False
+output_offset = 0
+
+############# templates
+basic_header = """
+#include <mmustructs.h>
+"""
+
+
+template_pd_complete = """
+static struct x86_mmu_page_directory page_directory = {
+REPACEMENT_HOOK
+};
+"""
+template_disabled_pde = """entry[PDE_NUMBER].pt = {.value = 0 }"""
+
+template_enabled_pde = """    .entry[PDE_NUMBER].pt = {
```

Comment:
I see value in having human-readable debugging output. But if I remember right, we can't generate the PDE in C because we can't actually populate the .page_table field with a valid memory address since it has to be shifted 12 bits.
Instead of outputting C, maybe just print out a human-readable table of the page directory / table contents to stdout? I think this is better than outputting C, someone might see the generated C code and try to build it.